Content.Server/Clothing/Systems/SkaterSystem.cs

```diff
@@ -0,0 +1,29 @@
+using Robust.Shared.Random;
+using Content.Shared.Movement.Events;
+using Content.Shared.Stunnable;
+using Content.Shared.Clothing;
+
+namespace Content.Server.Clothing;
+
+public sealed class SkaterSystem : EntitySystem
+{
+    [Dependency] private readonly SharedStunSystem _stunSystem = default!;
+    [Dependency] private readonly IRobustRandom _robustRandom = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<SkaterComponent, MoveInputEvent>(OnMove);
+    }
+
+    public void OnMove(EntityUid uid, SkaterComponent component, ref MoveInputEvent args)
```

Comment:
this could probably use inventory event relaying instead of adding/removing this component to the user, which would be more robust

Reply:
changed addcomp to ensurecomp

Reply:
If only i knew how to do this :)

Reply:
i think i fixed this, i also kinda cheated and removed the access restriction from two comps, if this isnt allowed then i gonna need help getting it to work.